### PR TITLE
Simplify flops calculation, apply for all kernel types

### PR DESCRIPTION
**Pull request by rengolin**
All tests pass, they check FLOPS calculations, so we know the new style is correct for all tested cases.

Now it also applies MLP FLOP calculations.

### CODE REVIEW

File: tools/mlir-gen/MLIRGen.cpp [hunk 1]
```diff
@@ -378,6 +351,16 @@ Value MLIRGenerator::lowerMatmul(MatMulArgs args) {
               })
           .getResult(0);
 
+  // Matmul flops = 2 * M * N * K = 2 * prod(inputDims) * N (weightShape[0])
+  int64_t mkFlops = 1;
+  for (int i = 0, max = inputShape.getRank(); i < max; i++)
+    mkFlops *= inputShape.getDimSize(i);
+  // Tiled: N = NB * n = weightShape[0] + weightShape[3]
+  int64_t nFlops = weightShape.getDimSize(0);
```
Comment (adam-smnk) [inline tools/mlir-gen/MLIRGen.cpp:359]: This grabs K dim if there's no tiling.

**Comment by rengolin** [comment]
> I like the idea but it gets a bit convoluted with complex tiling and non eltwise operations.

Thats why we do it by operation and not at the end. Once we compose everything, flops will always be correct.

Also, it's not super important to get the absolute perfection. Ball park is enough to give us what we want and this gives the exact same numbers as before for all tested cases.

File: tools/mlir-gen/MLIRGen.cpp (hunk 1 above)
Comment (rengolin) [reply tools/mlir-gen/MLIRGen.cpp:359]: ouch, ok. Not enough tests...

**Review by adam-smnk — APPROVED** [review]
Looks good - benchmark numbers are unchanged.
Thanks for adding extra test cases 👍

**Pull request merged**